client/extensions/zoninator/components/settings/zones-dashboard/index.jsx

```diff
@@ -4,58 +4,66 @@
 import React, { PropTypes } from 'react';
 import { connect } from 'react-redux';
 import { localize } from 'i18n-calypso';
-import { find, flowRight, get, noop } from 'lodash';
+import { find, flowRight, get, range } from 'lodash';
 
 /**
  * Internal dependencies
  */
-import Zone from './zone';
 import Button from 'components/button';
 import HeaderCake from 'components/header-cake';
 import SectionHeader from 'components/section-header';
 import sectionsModule from 'sections';
-import { getSelectedSiteSlug } from 'state/ui/selectors';
+import { getSelectedSiteId, getSelectedSiteSlug } from 'state/ui/selectors';
+import QueryZones from '../../data/query-zones';
+import ZoneItem from './zone-item';
+import ZonePlaceholder from './zone-placeholder';
+import { getZones, isFetchingZones } from '../../../state/zones/selectors';
 
-const zones = [
-	{ label: 'Foo', slug: 'foo', description: 'My first zone' },
-	{ label: 'Bar', slug: 'bar', description: 'Another zone' },
-	{ label: 'Baz', slug: 'baz', description: 'Another zone' },
-	{ label: 'Boo', slug: 'boo', description: 'Another zone' },
-];
+const placeholderCount = 16;
 
-const ZonesDashboard = ( { siteSlug, translate } ) => {
-	const getSettingsPath = () => {
-		const sections = sectionsModule.get();
-		const section = find( sections, ( value => value.name === 'zoninator' ) );
-
-		return get( section, 'settings_path' );
-	};
+const ZonesDashboard = ( { isFetching, siteId, siteSlug, translate, zones } ) => {
+	const sections = sectionsModule.get();
+	const section = find( sections, ( value => value.name === 'zoninator' ) );
+	const settingsPath = get( section, 'settings_path' );
 
 	return (
 		<div>
-			<HeaderCake backHref={ `/plugins/zoninator/${ siteSlug }` } onClick={ noop }>
+			<QueryZones siteId={ siteId } />
+
+			<HeaderCake backHref={ `/plugins/zoninator/${ siteSlug }` }>
 				Zoninator Settings
 			</HeaderCake>
 
 			<SectionHeader label={ translate( 'Zones' ) }>
-				<Button compact href={ `${ getSettingsPath() }/new/${ siteSlug }` }>
+				<Button compact href={ `${ settingsPath }/new/${ siteSlug }` }>
 					{ translate( 'Add a zone' ) }
 				</Button>
 			</SectionHeader>
-			{ zones.map( ( { label, slug, description } ) => (
-					<Zone key={ slug } label={ label } slug={ slug } description={ description } />
-				) ) }
+			{ isFetching && zones.length === 0 && range( placeholderCount ).map( i => (
+				<ZonePlaceholder key={ i } />
+			) ) }
+			{ zones.map( ( zone, idx ) => (
+				<ZoneItem key={ idx } zone={ zone } />
```

Comment:
Nit -- I prefer to avoid using `index`. In this case, `zone.slug` would make a unique candidate for `key`. (But that's just my taste, feel free to ignore 😉)